iroh-bytes/src/store/file.rs

```diff
@@ -0,0 +1,2483 @@
+        let progress = IgnoreProgressSender::default();
+        let (tag, _size) = self.finalize_import_sync(file, format, id, progress)?;
+        Ok(tag)
+    }
+
+    fn finalize_import_sync(
+        &self,
+        file: ImportSource,
+        format: BlobFormat,
+        id: u64,
+        progress: impl ProgressSender<Msg = ImportProgress> + IdGenerator,
+    ) -> OuterResult<(TempTag, u64)> {
+        let data_size = file.len()?;
+        tracing::info!("finalize_import_sync {:?} {}", file, data_size);
+        progress.blocking_send(ImportProgress::Size {
+            id,
+            size: data_size,
+        })?;
+        let progress2 = progress.clone();
+        let (hash, outboard) = match file.content() {
+            MemOrFile::File(path) => {
+                let span = trace_span!("outboard.compute", path = %path.display());
+                let _guard = span.enter();
+                let file = std::fs::File::open(path)?;
+                compute_outboard(file, data_size, move |offset| {
+                    Ok(progress2.try_send(ImportProgress::OutboardProgress { id, offset })?)
+                })?
+            }
+            MemOrFile::Mem(bytes) => {
+                // todo: progress? usually this is will be small enough that progress might not be needed.
+                compute_outboard(bytes, data_size, |_| Ok(()))?
+            }
+        };
+        progress.blocking_send(ImportProgress::OutboardDone { id, hash })?;
+        // from here on, everything related to the hash is protected by the temp tag
+        let tag = self.temp_tag(HashAndFormat { hash, format });
+        let hash = *tag.hash();
+        // blocking send for the import
+        let (tx, rx) = flume::bounded(1);
+        self.tx.send(ActorMessage::Import {
+            cmd: Import {
+                content: HashAndFormat { hash, format },
+                file,
+                outboard,
+                data_size,
+            },
+            tx,
+        })?;
+        Ok(rx.recv()??)
+    }
+
+    fn temp_file_name(&self) -> PathBuf {
+        self.path_options.temp_file_name()
+    }
+}
+
+impl Drop for StoreInner {
+    fn drop(&mut self) {
+        if let Some(handle) = self.handle.take() {
+            self.tx.send(ActorMessage::Shutdown).ok();
+            handle.join().ok();
+        }
+    }
+}
+
+struct ActorState {
+    handles: BTreeMap<Hash, BaoFileHandle>,
+    protected: BTreeSet<Hash>,
+    temp: Arc<RwLock<TempCounterMap>>,
+    msgs: flume::Receiver<ActorMessage>,
+    path_options: PathOptions,
+    inline_options: InlineOptions,
+    create_options: Arc<BaoFileConfig>,
+    rt: tokio::runtime::Handle,
+}
+
+/// The actor for the redb store.
+///
+/// It is split into the database and the rest of the state to allow for split
+/// borrows in the message handlers.
+#[derive(derive_more::Deref, derive_more::DerefMut)]
+struct Actor {
+    db: redb::Database,
+    #[deref]
+    #[deref_mut]
+    state: ActorState,
```

Comment:
Not too sure if I like the `deref` here - it caused a little irritation when seeing `self.msgs` or similar on the `impl Actor` - but maybe it's fine also, a comment wouldn't hurt though.